### PR TITLE
fix: returned block.chainId does not match provided SOLO_CHAIN_ID value

### LINKED ISSUE #4500: `eth_chainId` disagrees with EVM `block.chainid`
## Summary

On a Solo JSON-RPC relay configured with `eth_chainId = 1337` (example), a contract `eth_call` still sees **`block.chainid = 298`**.

| Source | Value |
|--------|--------|
| `eth_chainId` | `1337` (`0x539`) |
| `net_version` | `"1337"` |
| `block.chainid` (in contract execution) | **298** |

Anything that reads `eth_chainId` off-chain (wallets, scripts, tests) will disagree with on-chain logic that uses `block.chainid` (destination checks, chain-gated features, etc.).

## Actual (example from our setup)

```text
eth_chainId / net_version  → 1337
executionChainId()         → 298
```

## Impact

- Off-chain code using `eth_chainId` does not match on-chain `block.chainid`
- Breaks flows that compare `dstChainId` (or similar) to `block.chainid`
- Breaks any signing or config that assumes `eth_chainId` is the execution chain id (EIP-712 is one example; not required to demonstrate this bug)

---

## Suggested fix

Align relay `eth_chainId` / `net_version` with consensus `block.chainid`, or document the execution chain id that must be used everywhere instead of `eth_chainId`.

---

That keeps the repro to one contract, one deploy, and two `cast` calls.

### Describe the expected behavior

`eth_chainId` (and `net_version`, if applicable) equal `executionChainId()` / `block.chainid`.

## Minimal repro contract

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.28;

/// @notice Returns block.chainid as seen during EVM execution on this relay.
contract ChainIdProbe {
    function executionChainId() external view returns (uint256) {
        return block.chainid;
    }
}
```

---

### 1. Start one Solo relay

```
SOLO_CHAIN_ID=1337 npx @hiero-ledger/solo one-shot single deploy \
  -d clpr-solo-a-deployment \
  -q \
  -c clpr-solo-a-cluster \
  -n clpr-solo-a \
  --force-port-forward=false
```

```bash
export RPC=http://127.0.0.1:37546          # your relay URL
export DEPLOYER_PRIVATE_KEY=0x...        # funded account
```

### 2. Deploy the probe

```bash
forge init chainid-repro && cd chainid-repro
# save ChainIdProbe.sol to src/ChainIdProbe.sol

forge create src/ChainIdProbe.sol:ChainIdProbe \
  --rpc-url $RPC \
  --private-key $DEPLOYER_PRIVATE_KEY

export PROBE=0xYourDeployedAddress
```

### 3. Compare relay vs execution chain id

```bash
echo "Relay eth_chainId:"
cast chain-id --rpc-url $RPC

echo "EVM block.chainid:"
cast call $PROBE "executionChainId()(uint256)" --rpc-url $RPC
```

Optional raw JSON-RPC:

```bash
curl -s $RPC -H 'content-type: application/json' \
  -d '{"jsonrpc":"2.0","id":1,"method":"eth_chainId","params":[]}' | jq .

curl -s $RPC -H 'content-type: application/json' \
  -d '{"jsonrpc":"2.0","id":1,"method":"net_version","params":[]}' | jq .
```

---

### Additional Context

Solo version: 0.72.0

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This pull request changes the following:

* TBD

### Related Issues

* Closes #4500
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/commands/flags.ts [hunk 1]
```diff
@@ -896,7 +896,10 @@ export class Flags {
     name: 'chain-id',
     definition: {
       describe: 'Chain ID',
-      defaultValue: constants.HEDERA_CHAIN_ID, // Ref: https://github.com/hiero-ledger/hiero-json-rpc-relay#configuration
+      // Ref: https://github.com/hiero-ledger/hiero-json-rpc-relay#configuration
+      get defaultValue(): string {
```
Comment (jan-milenkov) [inline src/commands/flags.ts:900]: is this intentional? I've never seen this pattern before, I like the look of it

Comment (Ivo-Yankov) [reply src/commands/flags.ts:900]: Yes, it's intentional. It allows us to control the default value after the TypeScript files are initially loaded. In this exact case, I needed to test a custom chain ID with `solo one-shot single deploy`, but the command does not accept this flag, only the `SOLO_CHAIN_ID` env variable. So when I set that in `proces.env.SOLO_CHAIN_ID=xxxxx` in the e2e test, the constant `HEDERA_CHAIN_ID` was already loaded in memory with its default value `298`. 

The approach above makes sure that the default value is calculated when the flag value is requested, not at the initial load.

**Review by jan-milenkov — APPROVED** [review]
